### PR TITLE
Fix potential deadlock on Close()

### PR DESCRIPTION
**Pull request by danp60**
Fixes deadlock issue described in #123

### CODE REVIEW

**Comment by jared2501** [comment]
ping @arbll @jmoiron - would love to get this change looked at. I ran into this issue separately.

File: statsd/statsd.go [hunk 1]
```diff
@@ -331,6 +335,11 @@ func (c *Client) watch() {
 	for {
 		select {
 		case <-ticker.C:
+			// c.debugDeadlock is always nil except for the TestCloseDeadlock test.
+			if c.debugDeadlock != nil {
+				<-c.debugDeadlock
+			}
+
```
Comment (hush-hush) [inline statsd/statsd.go]: Could we instead overwrite the `NewTicker` function ? Like we do [here](https://github.com/DataDog/datadog-agent/blob/master/pkg/metadata/scheduler.go#L25).

That way the code logic is the same when running tests or not and we can control when the ticker triggers.

Comment (jared2501) [reply statsd/statsd.go]: @hush-hush - do you think you can just take this PR over in a separate branch and make the changes yourself? I am a datadog customer and rely on datadog-go to integrate my company's product into the datadog stack. This bug is preventing me from upgrading our dependency and taking advantage of new features.

Comment (hush-hush) [reply statsd/statsd.go]: After looking more into this I think it's fine as it is :+1:

**Comment by danp60** [comment]
Apologies, I haven't had time to look at the test logic feedback. What do I need to do to get this merged upstream?

**Comment by hush-hush** [comment]
Hi @danp60,

I cherry-picked your fix in #135. Thanks again for this fix ! This will be included in the next release.

**Pull request closed without merging**